mmv1/products/apigateway/terraform.yaml

```diff
@@ -26,15 +26,17 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         min_version: beta
         name: "apigateway_api_basic"
         primary_resource_id: "api"
+        primary_resource_name: "fmt.Sprintf(\"tf-test-api-%s\", context[\"random_suffix\"])"
         vars:
-          name: "api"
+          api_id: "api-"
```

Comment:
Note: dangling dash characters are here to trigger logic that automatically adds `tf-test-` at the start of the value in the templated HCL used in the test. Having a `-` present is necessary for this.

Reply:
Yeah that's a good idea, I'll push a commit making these vars have prefixes after discussion about the issue(s) with ServiceDirectory tests is concluded - I don't want to make the PR timeline more disjointed!

Reply:
👍 Alternately / additionally, I'd be happy to approve the straightforward bits in a separate PR while we sort out the complicated bits here.

Reply:
That sounds like the better solution - this PR ballooned a lot from the original intention

Reply:
[Here's the new PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/7185) - it contains everything in this PR and leaves servicedirectory and datacatalog untouched other than changes needed to let the new import steps in those tests build (pass correct number of args into fmt.Sprintf).

There are failing VCR tests for servicedirectory and datacatalog, but those would be addressed in this PR